Microsoft.Toolkit.Uwp.UI.Animations/Options/RepeatOption.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using Microsoft.Toolkit.Diagnostics;
+using Windows.Foundation.Metadata;
+using Windows.UI.Composition;
+using Windows.UI.Xaml.Media.Animation;
+
+#pragma warning disable CS0419
+
+namespace Microsoft.Toolkit.Uwp.UI.Animations
+{
+    /// <summary>
+    /// A type describing the repeat behavior for a custom animation.
+    /// </summary>
+    [CreateFromString(MethodName = "Microsoft.Toolkit.Uwp.UI.Animations.RepeatOption.Parse")]
+    public readonly struct RepeatOption
+    {
+        /// <summary>
+        /// The number of iterations for the animation.
+        /// </summary>
+        private readonly int value;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RepeatOption"/> struct.
+        /// </summary>
+        /// <param name="value">The number of iterations for the animation.</param>
+        private RepeatOption(int value)
+        {
+            this.value = value;
+        }
+
+        /// <summary>
+        /// Gets a <see cref="RepeatOption"/> value representing a single iteration.
+        /// </summary>
+        public static RepeatOption One => new(1);
+
+        /// <summary>
+        /// Gets a <see cref="RepeatOption"/> value indicating an animation that repeats forever.
+        /// </summary>
+        public static RepeatOption Forever => new(-1);
+
+        /// <summary>
+        /// Creates a <see cref="RepeatOption"/> value with the specified number of iterations.
+        /// </summary>
+        /// <param name="count">The number of iterations for the animation.</param>
+        /// <returns>A <see cref="RepeatOption"/> value with the specified number of iterations.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when <paramref name="count"/> is negative.</exception>
+        [Pure]
+        public static RepeatOption Count(int count)
+        {
+            Guard.IsGreaterThanOrEqualTo(count, 0, nameof(count));
+
+            return new(count);
+        }
+
+        /// <summary>
+        /// Parses a <see cref="RepeatOption"/> value from a <see cref="string"/>.
+        /// The allowed values are either non-negative integers, or "Forever".
+        /// </summary>
+        /// <param name="text">The input text to parse.</param>
+        /// <returns>The parsed <see cref="RepeatOption"/> value.</returns>
+        [Pure]
+        public static RepeatOption Parse(string text)
+        {
+            if (int.TryParse(text, out int count))
+            {
+                return Count(count);
+            }
+
+            if (text.AsSpan().Trim().SequenceEqual("Forever".AsSpan()))
```

Comment:
Does it matter that it's capitalized, is XAML usually case-sensitive? (I know stylistically this is the preferred way to capitalize it and as I would input it, just wondering if we should accept `forever` as well?)

Reply:
I'd say it depends - it's case sensitive for things such as XML tags, property names etc., but I just tried and it seems to be case insensitive for enum values (eg. `Visibility="visible"` works). We can make this insensitive too for consistency 🙂